### PR TITLE
Add capturable/maximize tests to Adam(W) optim configs

### PR DESCRIPTION
**Pull request by janeyx99**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104780
* #104779
* __->__ #104669

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Btw, can these be just using a foreach_lerp_?
```
        # Decay the first and second moment running average coefficient
        torch._foreach_mul_(device_exp_avgs, beta1)
        torch._foreach_add_(device_exp_avgs, device_grads, alpha=1 - beta1)

        torch._foreach_mul_(device_exp_avg_sqs, beta2)
        torch._foreach_addcmul_(device_exp_avg_sqs, device_grads, device_grads, 1 - beta2)
```

**Comment by janeyx99** [comment]
> Btw, can this be just using a `torch._foreach_lerp_`?
> 
> ```
>         # Decay the first and second moment running average coefficient
>         torch._foreach_mul_(device_exp_avgs, beta1)
>         torch._foreach_add_(device_exp_avgs, device_grads, alpha=1 - beta1)
> ```

Yes for the first. Opened https://github.com/pytorch/pytorch/pull/104781 to address more broadly.
No for the second since we want to avoid intermediates for peak memory sake.

**Review by albanD — APPROVED** [review]
:+1:

File: test/optim/test_optim.py [hunk 1]
```diff
@@ -776,14 +776,14 @@ def _test_derived_optimizers(self, optimizer_pairs_with_flags, flag):
     @property
     def _multi_tensor_optimizer_configs(self):
         return [
-            (optim.Adam, dict(weight_decay=1.0, amsgrad=True, fused=False)),
-            (optim.Adam, dict(weight_decay=1.0, amsgrad=False, fused=False)),
-            (optim.Adam, dict(weight_decay=0.0, amsgrad=True, fused=False)),
-            (optim.Adam, dict(weight_decay=0.0, amsgrad=False, fused=False)),
-            (optim.AdamW, dict(weight_decay=1.0, amsgrad=True)),
+            (optim.Adam, dict(weight_decay=1.0, amsgrad=True, capturable=True)),
```
[the remark this answers is not in the document]
Comment (janeyx99) [reply test/optim/test_optim.py]: Alright, we can add new ones

**Pull request closed without merging**